odoo/models.py

```diff
@@ -3120,12 +3123,14 @@ def _condition_to_sql(self, alias: str, fname: str, operator: str, value, query:
             sql_value = value
         elif need_wildcard:
             sql_value = SQL("%s", f"%{value}%")
+        elif field.translate:
+            sql_value = SQL("%s", field.convert_to_column(value, self, validate=False).adapted['en_US'])
         else:
             sql_value = SQL("%s", field.convert_to_column(value, self, validate=False))
 
         sql_left = sql_field
-        if operator.endswith('like'):
-            sql_left = SQL("%s::text", sql_field)
+        if operator.endswith('like') and field.type not in ('char', 'text', 'html'):
+            sql_left = SQL("(%s)::text", sql_field)
```

Comment:
Adding the `::text` conversion for non-text fields only makes the generated SQL much better :+1: